app/controllers/application_controller.rb

```diff
@@ -70,7 +70,7 @@ def after_sign_in_path_for(resource_or_scope)
   end
 
   def redirect_user_back_after_login
-    if request.env['REQUEST_URI'].present? && !request.xhr? && session[:return_to].nil?
+    if request.env['REQUEST_URI'].present? && !request.xhr?
```

Comment:
won't this break the fallback cases? I believe we do not have specs for them. Send a feature spec for a fallback case (like logging in for the contribution through the get fallback route).